### DISCUSSION TITLE
Not finding the GDAS elevation file during LDT “Ensemble restart processing”

### DISCUSSION THREAD

**Discussion by Gayathridevi123** [NASA-LIS/LISF#1061] [Q&A] [answered]
Hi,

I'm trying to generate an ensemble restart file for data assimilation. Forcing is GDAS and LSM is Noah3.9. But while running LDT “Ensemble restart processing” it is not finding the GDAS elevation file:-

 Reading the GDAS forcing data
 GDAS forcing directory :./input/MET_FORCING/GDAS
 [INFO] Initializing GDAS grid-1 for: 1991-2000 grid (T126)
 [INFO] Initializing GDAS grid-2 for: 2000-2002 grid (T170)
 [INFO] Initializing GDAS grid-3 for: 2002-2005 grid (T254)
 [INFO] Initializing GDAS grid-4 for: 2005-2010 grid (T382)
 [INFO] Initializing GDAS grid-5 for: 2010-2015 grid (T574)
 [INFO] Initializing GDAS grid-6 for: 2015-present grid (T1534)
  Projection and Transform: gaussian, bilinear
 **Reading forcing elev for GDAS_T126
 GDAS elevation file missing:                                                                                                    
 Opening/reading GDAS elev file::
 [ERR] error grib_open_file in read_gdas_elev Stopping.         -11
 endrun is being called**

In the terminal it is showing **ECCODES ERROR:  IO ERROR: Bad address: (null) (Bad address)**
Wondering why this is happening. For LDT "LSM parameter processing" the same files have been read by LDT and LIS open-loop outputs are generated.

Could anyone help me with this? Please find the ldt.config, ldtlog and lis_input_noah39.d01.nc files.
[ldt.config.txt](https://github.com/NASA-LIS/LISF/files/8505288/ldt.config.txt)
[ldtlog.txt](https://github.com/NASA-LIS/LISF/files/8505289/ldtlog.txt)
[lis_input_noah39.d01.txt](https://github.com/NASA-LIS/LISF/files/8505290/lis_input_noah39.d01.txt)

Thank you,
Gayathri

**Comment by bmcandr** [comment, marked as answer]
Hi @Gayathridevi123,

Based on your ldtlog file it looks like the elevation filepath itself is not being read into LDT properly which naturally fails a "does this file exist?" check. Here is what the log should contain when initializing GDAS, note the presence of the filename:

```
...
 Reading forcing elev for GDAS_T126
 Opening/reading GDAS elev file::
 ./input/LS_PARAMETERS/metforcing_parms/GDAS/global_orography.t126.grb
 #Regridding: Applying bilinear interp to input parameter
 Done reading forcing elevation field.
...
```

What is odd is that we check that _some_ path is present in the config file when reading config entries ([see here](https://github.com/NASA-LIS/LISF/blob/0e4b2ec81d1a2fa93d44d1a5efd910bf2e188630/ldt/metforcing/gdas/readcrd_gdas.F90#L53-L57)). This should fail if no path is present, causing LDT to exit. Your GDAS entries are identical to the [ldt.config in our GDAS testcase](https://github.com/NASA-LIS/LISF/blob/0e4b2ec81d1a2fa93d44d1a5efd910bf2e188630/lis/testcases/metforcing/gdas/ldt.config#L89-L94) so I don't see a problem there.

Let's start troubleshooting this by modifying lines 53-57 in `ldt/metforcing/gdas/readcrd_gdas.F90` to print another log message containing the filepath just after LDT reads it:

```fortran
    call ESMF_ConfigFindLabel(LDT_config,"GDAS T126 elevation map:",rc=rc)
    do n=1,LDT_rc%nnest
       call ESMF_ConfigGetAttribute(LDT_config,LDT_rc%gdasT126elevfile(n),rc=rc)
       call LDT_verify(rc,'GDAS T126 elevation map: not specified')
       write(LDT_logunit,*) "T126 filepath: ", LDT_rc%gdasT126elevfile(n)
    end do
```

Re-compile, try to run again, and report back the contents of your log.

----

I do see there are some tab characters in your ldt.config file. See the `^I` characters in the screenshot below:

![Screen Shot 2022-04-18 at 16 49 10](https://user-images.githubusercontent.com/19274445/163875685-71a8ad26-de32-4f56-bb38-b1367a66b221.png)

I don't _think_ these would cause a problem given other paths are read, but it couldn't hurt to replace these with spaces. To see these characters open the file in `vim` and type `:set list`.

**Reply by Gayathridevi123** [reply]
Hi [bmcandr](https://github.com/bmcandr),

Thank you for your reply. I need to point out 

1. I'm using the master branch of LISF. So the mentioned lines (lines 53-57 in ldt/metforcing/gdas/readcrd_gdas.F90) are same. Also I noted 51th line "**LDT_rc%runmode == "LSM parameter processing" ) then**". Here run mode is LSM parameter processing but I'm trying to generate the Ensemble restart file.

2.  GDAS Files exist and are there in the path mentioned.

3.  Thank you for mentioning the tab characters (^I), I removed those.

4.  LDT runs with the "**LSM Parameter Processing**" option. The same config file, if I changed it to "LSM Parameter Processing" option those GDAS elevation files are read by LDT successfully (ldt.config and ldt.log are attached).
[ldt.config.txt](https://github.com/NASA-LIS/LISF/files/8509975/ldt.config.txt)
[ldtlog1.txt](https://github.com/NASA-LIS/LISF/files/8509978/ldtlog1.txt)
As I mentioned earlier, LDT "LSM parameter processing" the same GDAS files have been read by LDT and LIS open-loop outputs are generated.

**5. Issue is when changing it to "**Ensemble restart processing**". When given this option it is not finding the GDAS files. I'm trying to run on the same directory in which LDT runs successfully with the LSM Parameter Processing option**.

Thank you
Gayathri

**Reply by bmcandr** [reply]
Hi @Gayathridevi123,

Thanks for the clarification regarding the runmode you are using.

Can you try setting `Number of met forcing sources: 0` to disable metforcing source? As far as I can metforcing data do not need to be processed for upscaling/downscaling LSM restart files. Here is an example ldt.config file for this runmode: [ldt.config_ensrst.txt](https://github.com/NASA-LIS/LISF/files/8513740/ldt.config_ensrst.txt)

Note that metforcing is disabled in this config file.

@karsenau, should/can metforcing sources be disabled when using LDT's `Ensemble restart processing` runmode?

**Reply by Gayathridevi123** [reply]
Hi [bmcandr](https://github.com/bmcandr),

Thank you for the config file. By setting Number of met forcing sources: 0 LDT ran successfully.

I just want to clarify about the metforcing data.  Is it not needed for ensemble restart file generation?

Thanks
Gayathri

**Reply by bmcandr** [reply]
I am not very well versed in this area of LISF, but my understanding is that LDT is using an existing LIS restart file to generate a new restart file. It is not creating a new LIS input parameter file and therefore the metforcing inputs are not needed for this runmode.

**Reply by Gayathridevi123** [reply]
Thank you Brendan for your help.

**Reply by karsenau** [reply]
Yes -- to confirm that you want to set the number of met forcing sources to "0" when working on generating a new restart file using LDT.  The restart file only cares about the LSM or hydrological model states that are needed to initialize or restart the same model runs.
